lib/active_merchant/billing/gateways/payeezy.rb

```diff
@@ -180,50 +232,63 @@ def add_soft_descriptors(params, options)
       end
 
       def commit(params, options)
-        url = if options[:integration]
-          integration_url
-        elsif test?
-          test_url
-        else
-          live_url
-        end
+        url = base_url(options) + endpoint(params)
 
         if transaction_id = params.delete(:transaction_id)
           url = "#{url}/#{transaction_id}"
         end
 
         begin
-          body = params.to_json
-          response = parse(ssl_post(url, body, headers(body)))
+          response = api_request(url, params)
         rescue ResponseError => e
           response = response_error(e.response.body)
         rescue JSON::ParserError
-          response = json_error(raw_response)
+          response = json_error(e.response.body)
         end
 
         Response.new(
           success_from(response),
           handle_message(response, success_from(response)),
           response,
           test: test?,
-          authorization: authorization_from(params, response),
+          authorization: success_from(response) ? authorization_from(params, response) : nil,
           avs_result: {code: response['avs']},
           cvv_result: response['cvv2'],
           error_code: error_code(response, success_from(response))
         )
       end
 
-      def success_from(response)
-        response['transaction_status'] == 'approved'
+      def base_url(options)
+        if options[:integration]
+          integration_url
+        elsif test?
+          test_url
+        else
+          live_url
+        end
       end
 
-      def authorization_from(params, response)
-        [
-          response['transaction_id'],
-          response['transaction_tag'],
-          params[:method],
-          (response['amount'] && response['amount'].to_i)
-        ].join('|')
+      def endpoint(params)
+        params[:ta_token].present? ? '/securitytokens' : '/transactions'
+      end
+
+      def api_request(url, params)
+        if params[:ta_token].present?
```

Comment:
Hmm. Checking if `transaction_type` is `store` might be more readable/reasonable?

Reply:
No `transaction_type` in calls to the `/securitytokens` endpoint.

Reply:
As I'm reading through this and run into `params[:ta_token]` used in a few conditionals I start to wonder what the presence of a `ta_token` represents. It seems that one must know that a `ta_token` is used when storing a card. I came to this conclusion by scrolling up and reading `add_creditcard_for_tokenization`. This is fine, but I wonder if we should encapsulate this concept in a single place, i.e. (another) private method.

Reply:
👍 Great suggestion -- added and open to naming suggestions.